tests/phpunit/CRM/Utils/ArrayTest.php

```diff
@@ -367,4 +367,52 @@ public function testFlatten() {
     $this->assertEquals($flat, $expected);
   }
 
+  public function testFormatArrayKeysTransformsAssociativeArrayToSequential() {
+    $tests = [
```

Comment:
instead of having an array inside this method like that with all the test cases, you can use something called data provider : https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html

Reply:
This will produce a second function, we are testing a specific method and we will not need this data outside this method ever. So I think keeping it inside might be a good idea in this particular case. This is a bit different from if we had, for example, a collection of contacts data that we would reuse very often.

Reply:
@igorpavlov the advantage of data providers over iterating within a function is that if you have 8 test cases & one fails then the whole test will fail - but with a data provider it will still try the other 7